File: process_logo.mjs
import { Jimp } from 'jimp';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function processLogo() {
    const inputPath = path.join(__dirname, 'public', 'khrunch-icon-source.png');
    const outputPath = path.join(__dirname, 'public', 'khrunch-icon.png');

    console.log(`Processing ${inputPath}...`);

    try {
        // Try to read
        console.log("Reading image...");
        const image = await Jimp.read(inputPath);
        console.log("Image read successfully.");

        // Get corner color (assuming it's the background)
        const cornerColor = image.getPixelColor(0, 0);
        // Manual intToRGBA
        const r = (cornerColor >>> 24) & 0xFF;
        const g = (cornerColor >>> 16) & 0xFF;
        const b = (cornerColor >>> 8) & 0xFF;
        const a = cornerColor & 0xFF;
        console.log(`Corner color: rgba(${r}, ${g}, ${b}, ${a})`);

        console.log("Scanning pixels...");
        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
            const red = this.bitmap.data[idx + 0];
            const green = this.bitmap.data[idx + 1];
            const blue = this.bitmap.data[idx + 2];

            // Check for white
            if (red > 240 && green > 240 && blue > 240) {
                this.bitmap.data[idx + 3] = 0; // Transparent
            }
            // Check for gray checkerboard
            else if (Math.abs(red - green) < 10 && Math.abs(red - blue) < 10 && (red > 190 && red < 240)) {
                this.bitmap.data[idx + 3] = 0; // Transparent
            }
            // Also check against corner color
            else if (red === r && green === g && blue === b) {
                this.bitmap.data[idx + 3] = 0;
            }
        });

        console.log("Autocropping...");
        image.autocrop();

        console.log("Writing output...");
        // Try writeAsync if available, else write
        if (image.writeAsync) {
            await image.writeAsync(outputPath);
        } else {
            await image.write(outputPath);
        }
        console.log(`Saved to ${outputPath}`);

    } catch (error) {
        console.error('Error processing image:', error);
        if (error.message) console.error(error.message);
        if (error.stack) console.error(error.stack);
    }
}

processLogo();

File: process_logo.py
from PIL import Image
import numpy as np

def process_logo(input_path, output_path):
    print(f"Processing {input_path}...")
    try:
        img = Image.open(input_path)
        img = img.convert("RGBA")
        datas = img.getdata()
        
        newData = []
        # Checkerboard colors often used:
        # White: (255, 255, 255)
        # Light Gray: (204, 204, 204) or similar
        # Let's sample the corners to be sure
        width, height = img.size
        corners = [
            img.getpixel((0, 0)),
            img.getpixel((width-1, 0)),
            img.getpixel((0, height-1)),
            img.getpixel((width-1, height-1))
        ]
        print(f"Corner pixels: {corners}")
        
        # It seems the user uploaded a screenshot of a transparent png, so it has the checkerboard.
        # We need to make the checkerboard transparent.
        # A simple heuristic: if a pixel is white or the light gray of the checkerboard, make it transparent.
        # But we must be careful not to remove parts of the logo if they are the same color.
        # The logo is Orange and Dark Gray.
        
        # Let's try to detect the content bounding box first by excluding the background colors.
        # Assuming background is the checkerboard.
        
        # We can also just use a flood fill from the corners if the logo is contained?
        # No, the text might have holes.
        
        # Let's iterate and replace.
        # Common checkerboard gray is often around 204 (0xCC) or 238 (0xEE).
        
        # Let's look at the unique colors in the image to identify the checkerboard.
        # Or just use the corner colors as the "background" set.
        
        background_colors = set(corners)
        # Add common checkerboard variations just in case
        background_colors.add((255, 255, 255, 255)) # White
        background_colors.add((204, 204, 204, 255)) # Gray
        
        print(f"Treating these as background: {background_colors}")

        for item in datas:
            if item in background_colors:
                newData.append((255, 255, 255, 0))
            else:
                # Also check for "near" background colors (antialiasing edges of the checkerboard?)
                # For now, exact match.
                newData.append(item)

        img.putdata(newData)
        
        # Now crop to content
        bbox = img.getbbox()
        if bbox:
            print(f"Cropping to {bbox}")
            img = img.crop(bbox)
        else:
            print("No content found!")

        img.save(output_path, "PNG")
        print(f"Saved to {output_path}")
        
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    process_logo("public/khrunch-logo-v2.png", "public/khrunch-logo-transparent.png")
